File: translation_output/starcoder/codenet/java/python-sanitized/p04019.py
import java.util.Scanner;

public class p04019 {

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		String s = scanner.next();
		boolean checkS = False;
		boolean checkN = False;
		boolean checkW = False;
		boolean checkE = False;
		for(int i=0;i<s.length();i++){
			if(s.charAt(i) == 'S'){
				checkS = True;
			}else if(s.charAt(i) == 'N'){
				checkN = True;
			}else if(s.charAt(i) == 'W'){
				checkW = True;
			}else if(s.charAt(i) == 'E'){
				checkE = True;
			}
		}
		if(checkS == checkN and checkW == checkE):
			System.out.println("Yes");
		else:
			System.out.println("No");
	}

}

File: Results/DER/Translation/translation_output/misleading_test/gpt-4/codenet/python/java-sanitized/p04019.java
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class p04019 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.next();
        scanner.close();

        Set<Character> uniqueDirections = new HashSet<>();
        for (char ch : input.toCharArray()) {
            uniqueDirections.add(ch);
        }

        if (uniqueDirections.size() == 4) {
            System.out.println("Yes");
        } else if (uniqueDirections.size() == 3) {
            System.out.println("No");
        } else if (uniqueDirections.size() == 2) {
            if (uniqueDirections.contains('N') && uniqueDirections.contains('S')) {
                System.out.println("Yes");
            } else if (uniqueDirections.contains('E') && uniqueDirections.contains('W')) {
                System.out.println("Yes");
            } else {
                System.out.println("No");
            }
        } else {
            System.out.println("No");
        }
    }
}